### ISSUE TITLE
Unable to load class 'org.jetbrains.kotlin.incremental.IncrementalJvmCompilerRunnerKt'

### ISSUE THREAD

**Issue by JordanLongstaff** [autonomousapps/dependency-analysis-gradle-plugin#944]
**Plugin version**
1.21.0

**Gradle version**
8.0

**(Optional) Android Gradle Plugin (AGP) version**
8.1.0

**Describe the bug**
After upgrading the plugin to version 1.21.0, my Kotlin compiler tasks all fail with this output:

```
Unable to load class 'org.jetbrains.kotlin.incremental.IncrementalJvmCompilerRunnerKt'.

This is an unexpected error. Please file a bug containing the idea.log file.
```

The dependency analysis tasks as well as the lint tasks are also failing with the same error.

**To Reproduce**
1. Upgrade the plugin to version 1.21.0.
2. Run the `projectHealth` task. The `compileKotlin` subtask will report the error.

**Expected behavior**
These tasks worked just fine with version 1.20.0. They shouldn't be failing.

**Additional context**
[idea.log](https://github.com/autonomousapps/dependency-analysis-android-gradle-plugin/files/12381748/idea.log)

**Comment by kyhule** [comment]
I noticed the same issue on my personal project. Here is a run upgrading the build plugin in which DAGP was upgraded to 1.2.1 https://github.com/kyhule/polyworld-app/actions/runs/5933064716/job/16087936262#step:6:47.

**Comment by autonomousapps** [comment]
Thanks for the report. Can you update your version of Gradle to latest, which is 8.3 at time of writing?

**Comment by kyhule** [comment]
That build was with gradle 8.3 though it was not with kotlin 1.9.0. I think this might be caused by DAGP including the kotlin bom for 1.9.0. It seems to cause a mixed bag of kotlin versions on the build classpath.

**Comment by kyhule** [comment]
After upgrading to kotlin 1.9.0, the issue seems to have gone away.
https://github.com/kyhule/polyworld-app/actions/runs/5957791886/job/16161054676
